module/solutions/helper.js

```diff
@@ -2659,37 +2705,93 @@ module.exports = class SolutionsHelper {
 		userId,
 		pageSize,
 		pageNo,
-		search,
+		search = '',
 		filter,
 		surveyReportPage = '',
-		currentScopeOnly = false,
-		entityId = ''
+		currentScopeOnly = false
 	) {
 		return new Promise(async (resolve, reject) => {
 			try {
 				currentScopeOnly = UTILS.convertStringToBoolean(currentScopeOnly)
-				let assignedSolutions = await this.assignedUserSolutions(
-					solutionType,
-					userId,
-					search,
-					filter,
-					'',
-					'',
-					entityId
-				)
-				if (!assignedSolutions.success) {
+				if (currentScopeOnly && process.env.SUBMISSION_LEVEL == 'ENTITY') {
```

Comment:
@Prajwal17Tunerlabs can't we throw this error in validator or controller?

Reply:
above piece of code moved to controller.